### ISSUE TITLE
kube-proxy IPVS tries to load nf_conntrack_ipv4 on RedHat 8.3+

### ISSUE THREAD

**Issue by champtar** [kubernetes/kubernetes#97052]
**What happened**:
kube-proxy IPVS tries to load nf_conntrack_ipv4 on RedHat 8.3+ (so soon on CentOS ;) )
RedHat 8.3 has kernel 4.18 with many backports, including nf_conntrack_ipv4 removed/merged into nf_conntrack
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.3_release_notes/new-features#enhancement_networking

**What you expected to happen**:
kube-proxy (and kubernetes components in general) do not use kernel versions to influence behavior

**How to reproduce it (as minimally and precisely as possible)**:
Run Kubernetes on RedHat 8.3+ or CentOS streams

**Anything else we need to know?**:
This is just a warning, not a blocker
```
# kubectl logs -n kube-system pod/kube-proxy-8w2tx
W1203 12:48:28.105739       1 proxier.go:635] Failed to load kernel module nf_conntrack_ipv4 with modprobe. You can ignore this message when kube-proxy is running inside container without mounting /lib/modules
W1203 12:48:28.132811       1 proxier.go:635] Failed to load kernel module nf_conntrack_ipv4 with modprobe. You can ignore this message when kube-proxy is running inside container without mounting /lib/modules
```

**Environment**:
- Kubernetes version (use `kubectl version`): v1.18.10
- Cloud provider or hardware configuration: VMs
- OS (e.g: `cat /etc/os-release`): Red Hat Enterprise Linux 8.3 (Ootpa)
- Kernel (e.g. `uname -a`): 4.18.0-240.1.1.el8_3.x86_64
- Install tools: Kubespray

**Comment by rikatz** [comment]
Hello @champtar 

Can you, anyway confirm that when the other modules are loaded?

Per the [code](https://github.com/kubernetes/kubernetes/blob/38f5dc8e7b75c16db29105b7c2314bfd3100840d/pkg/util/ipvs/ipvs.go#L129) yes, it checks if the kernel is less then 4.19 and loads the "nf_conntrack_ipv4" so this needs to be reverified.

**Comment by rikatz** [comment]
Got it :)

I'll take a look into the module loader to see if trying to load one then other and raising the error if both fails is a better approach ;)

**Comment by champtar** [comment]
I've looked at some systems (EL7, 8.2, 8.3), and ip_vs always depends on nf_conntrack

**Comment by rikatz** [comment]
@champtar opened the PR https://github.com/kubernetes/kubernetes/pull/97070

Can you please take a look?

**Comment by dougsland** [comment]
@rikatz @aojea looks like solved by kubespray?

**Comment by champtar** [comment]
To be clear kube-proxy is working just fine, with a warning. kubespray needed some love (merged but not released)

**Comment by dougsland** [comment]
> To be clear kube-proxy is working just fine, with a warning. kubespray needed some love (merged but not released)

Great. Can we close this one?

**Comment by champtar** [comment]
It'll get closed if/when #97070 is merged

**Comment by dougsland** [comment]
> It'll get closed if/when #97070 is merged

Oh @champtar, I though you said this report was about  kubespray only. Let me check this PR.
